orchestrator/crates/orchestrator/src/utils/logging.rs

```
@@ -0,0 +1,118 @@
+        event.record(&mut visitor);
+
+        if !visitor.meta.is_empty() {
+            write!(writer, " ({}{}{})", msg_color, visitor.meta, reset)?;
+        }
+
+        write!(writer, "{}{}{}", msg_color, visitor.message, reset)?;
+        if !visitor.fields.is_empty() {
+            write!(writer, " ({}{}{})", msg_color, visitor.fields, reset)?;
+        }
+
+        writeln!(writer)
+    }
+}
+
+// Visitor to extract message and format fields
+#[derive(Default)]
+struct FieldExtractor {
+    message: String,
+    fields: String,
+    meta: String,
+}
+
+impl tracing::field::Visit for FieldExtractor {
+    fn record_debug(&mut self, field: &tracing::field::Field, value: &dyn std::fmt::Debug) {
+        if field.name() == "message" {
+            self.message = format!("{:?}", value).trim_matches('"').to_string();
+        } else if field.name() == "yak" {
+            if !self.meta.is_empty() {
+                self.meta.push_str(", ");
+            }
+            self.meta.push_str(&format!("{}={:?}", field.name(), value));
+        } else {
+            if !self.fields.is_empty() {
+                self.fields.push_str(", ");
+            }
+            self.fields.push_str(&format!("{}={:?}", field.name(), value));
+        }
+    }
+}
+
+/// Initialize the tracing subscriber with
+/// - PrettyFormatter for console readability
+/// - JsonFormatter for json logging (for integration with orchestrator)
+///
+/// This will also install color_eyre to handle the panic in the application
+pub fn init_logging(service_name: &str) {
+    color_eyre::install().expect("Unable to install color_eyre");
+    LogTracer::init().expect("Failed to set logger");
+    let subscriber = fmt::Subscriber::builder()
+        .with_thread_names(true)
+        .with_thread_ids(true)
+        .with_target(false)
+        .event_format(PrettyFormatter::new(service_name))
+        .finish();
+
+    tracing::subscriber::set_global_default(subscriber).unwrap();
```

Either:

- replace this `unwrap` with an `expect`, or
- call `try_init` instead.

The latter can be quite important if this is called anywhere else than in `main`, as it could be overriding the default subscriber we are setting for tests. I would therefore suggest the latter option (we don't need to return the error, since if this fails it should mean that another subscriber has already been registered).